src/entities/DFGVariable.jl

```diff
@@ -104,11 +110,47 @@ timestamp(v::DFGVariable) = v.timestamp
 tags(v::DFGVariable) = v.tags
 estimates(v::DFGVariable) = v.estimateDict
 estimate(v::DFGVariable, key::Symbol=:default) = haskey(v.estimateDict, key) ? v.estimateDict[key] : nothing
+"""
+    $SIGNATURES
+
+Retrieve solver data structure stored in a variable.
+"""
 solverData(v::DFGVariable, key::Symbol=:default) = haskey(v.solverDataDict, key) ? v.solverDataDict[key] : nothing
-getData(v::DFGVariable; solveKey::Symbol=:default)::VariableNodeData = v.solverDataDict[solveKey]
+"""
+    $SIGNATURES
+
+Retrieve data structure stored in a variable.
+"""
+function getData(v::DFGVariable; solveKey::Symbol=:default)::VariableNodeData
+  @warn "getData is deprecated, please use solverData()"
+  return v.solverDataDict[solveKey]
+end
+"""
+    $SIGNATURES
+
+Set solver data structure stored in a variable.
+"""
 setSolverData(v::DFGVariable, data::VariableNodeData, key::Symbol=:default) = v.solverDataDict[key] = data
 solverDataDict(v::DFGVariable) = v.solverDataDict
 internalId(v::DFGVariable) = v._internalId
 # Todo: Complete this.
 smallData(v::DFGVariable) = v.smallData
 bigData(v::DFGVariable) = v.bigData
+
+"""
+    $(SIGNATURES)
+Structure for first-class citizens of a DFGVariable.
+"""
+mutable struct DFGVariableSummary <: AbstractDFGVariable
```

Comment:
we're missing the `softtypename::Symbol` here -- is that in a future PR?  @Affie did you have strong preference here for visualization.  We did talk about this a bit.